app/controllers/works_controller.rb

```diff
@@ -147,11 +147,18 @@ def review
     end
   end
 
-  def completed
+  def validate
     @work = Work.find(params[:id])
     @work.submission_notes = params["submission_notes"]
-    @work.complete_submission!(current_user)
-    redirect_to user_url(current_user)
+    @uploads = @work.uploads
+    @wizard_mode = true
+    begin
+      @work.complete_submission!(current_user)
+      redirect_to user_url(current_user)
+    rescue AASM::InvalidTransition
+      # Work couldn't transition, send the user back to the edit page
+      render "edit", status: :unprocessable_entity
+    end
```

Comment:
Notice the specific handle for validate renders the Edit page (so the user can fix the errors)

Reply:
Note, you would not need to repeat this code here.  In a PR earlier this week I broke out a different location for errors during the creation: https://github.com/pulibrary/pdc_describe/blob/main/app/controllers/works_controller.rb#L284-L290

Reply:
Good idea, done.